all/entrypoint.sh

```diff
@@ -184,6 +186,9 @@ plugins=grpc+embedded\
     "node")
         GEN_STRING="--grpc_out=$OUT_DIR --js_out=import_style=commonjs,binary:$OUT_DIR --plugin=protoc-gen-grpc=`which grpc_${PLUGIN_LANG}_plugin`"
         ;;
+    "node-ts")
+        GEN_STRING="--grpc_out=$OUT_DIR --js_out=import_style=commonjs,binary:$OUT_DIR --ts_out=$OUT_DIR --plugin=protoc-gen-grpc=`which grpc_${PLUGIN_LANG}_plugin`"
```

Comment:
the ts-protoc-gen plugin also supports generating gRPC stubs using grpc-web: https://github.com/improbable-eng/ts-protoc-gen#generating-grpc-service-stubs-for-use-with-grpc-web

I wondered if `node-ts` should drop the `protoc-gen-grpc` plugin and instead include the `service=true` parameter in `ts_out`?

Reply:
Note that there are 2 'grpc-web' implementations: Improbable's and Googles. ts-protoc-gen only generates stubs for use with Improbable's implementation.

Reply:
Thanks Jonny, and thanks for the new release!

So if I understand correctly, what’s in this PR would generate TypeScript declarations for protobuf messages and Google grpc-web service stubs without TypeScript declarations.

Then a second option could be to instead generate Improbable grpc-web service stubs with TypeScript declarations.

I’m curious if you have a recommendation?

Seems like it could be nice to fit in with the rest of this project using Google grpc-web, but that it’d also be nice for “node-ts” to generate TypeScript declarations for all the generated code?

(I was just reading https://github.com/improbable-eng/grpc-web/issues/199 and https://grpc.io/blog/state-of-grpc-web)

Reply:
I'd be open to add explicit options for node. That way, you'd have `-l node` and something like `--with-typescrypt` `--with-improbable` or `--with-ts-protoc-gen`

Thoughts?

Reply:
Nice, yeah I like that 👍

Reply:
I tried adding a `--with-typescript` option that only applies to `node` (and updated the PR description). Thanks for that suggestion - let me know what you think, whenever you have a chance to check it out!